File: Online Judges/LeetCode/30-day-leetcoding-challenge/Week-2/Contiguous Array.cpp
class Solution
{
public:
    int findMaxLength(vector<int> &nums)
    {
        map<int, int> mp;
        int len = 0, count = 0;
        mp[0] = -1;
        for (int i = 0; i < nums.size(); i++)
        {
            count = count + (nums[i] ? 1 : -1);
            auto fnd = mp.find(count);
            if (fnd != mp.end())
            {
                len = (i - mp[count]) > len ? i - mp[count] : len;
            }
            else
                mp[count] = i;
        }
        return len;
    }
};

File: Online Judges/LeetCode/May LeetCoding Challenge/Week-4/Contiguous Array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        count, maxLen = 0, 0
        mapTable = {0: 0}

        for index, num in enumerate(nums, 1):
            if num == 0:
                count -= 1
            else:
                count += 1
            if count in mapTable:
                maxLen = max(maxLen, index - mapTable[count])
            else:
                mapTable[count] = index
        return maxLen
